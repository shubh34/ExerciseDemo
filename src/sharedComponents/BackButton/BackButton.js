/* eslint-disable max-len */
import './BackButton.scss';
import React from 'react';
import PropTypes from 'prop-types';

const BackButton = ({ onClick, title = 'Back' }) => (
	<div className="back-button">
		<button type="button" name="go back" onClick={onClick}>{title}</button>
	</div>

);
BackButton.propTypes = {
	onClick: PropTypes.string.isRequired,
	title: PropTypes.string,
};

export default BackButton;
