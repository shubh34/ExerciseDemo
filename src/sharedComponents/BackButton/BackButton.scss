.back-button {
    z-index: 110;
    position: absolute; 
   & :before {
       content: '\e628';
       font-family: Glyphs;
       font-size: 32px;
   };
   button {
    font-size: 24px;
    padding: 16px;
    background: transparent;
    border: none;
    color:white;
    cursor: pointer;
   }
}