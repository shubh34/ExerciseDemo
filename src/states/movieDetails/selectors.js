import get from 'lodash/get';
import { createSelector } from 'reselect';

const getMovieDetails = state => get(state, 'movieDetails', '');

export const getHighLightedScore = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.scores', []).find(score => score.highlighted));

export const getMovieTitle = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.title', ''));

export const getSnapshotImageUrl = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.images.snapshot', '').replace('.jpeg', '-width1920-quality80.jpeg'));

export const getMovieDescription = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.plot', ''));

export const getScore = createSelector(getHighLightedScore, scores => get(scores, 'score', ''));

export const getVotes = createSelector(getHighLightedScore, scores => get(scores, 'formatted_amount_of_votes', ''));
