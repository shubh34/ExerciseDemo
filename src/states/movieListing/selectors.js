import get from 'lodash/get';

export const getMovieListHeader = (state, id) => get(state, `movieListing[${id}].data.name`);

export const getMovieListByCategory = (state, id) => get(state, `movieListing[${id}].data.contents.data`, []).map(movie => ({
	artwork: movie.images.artwork.replace('.jpeg', '-width217-quality80.jpeg'),
	id: movie.id,
	score: movie.highlighted_score.score,
	votes: movie.highlighted_score.formatted_amount_of_votes,
}));
