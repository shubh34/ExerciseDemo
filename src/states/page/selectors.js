export const getCurrentPage = state => state.page.name;
export const getBackPage = state => state.page.backPage;
