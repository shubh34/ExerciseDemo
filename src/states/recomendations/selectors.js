import get from 'lodash/get';
import { createSelector } from 'reselect';

export const getRecommendedMovies = state => get(state, 'recomendations.data', []);

export const getFormattedRecomendedMovieDetails = createSelector(
	getRecommendedMovies,
	movies => movies.map(movieDetails => ({
		id: movieDetails.id,
		artwork: movieDetails.images.artwork.replace('.jpeg', '-width217-quality80.jpeg'),
		score: movieDetails.highlighted_score.score,
		votes: movieDetails.highlighted_score.formatted_amount_of_votes
	}))
);
