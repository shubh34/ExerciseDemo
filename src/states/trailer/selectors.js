import get from 'lodash/get';

export const getMovieTrailerUrl = state => get(state, 'trailer.trailer.data.stream_infos[0].url');
