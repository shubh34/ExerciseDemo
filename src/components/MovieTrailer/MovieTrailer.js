/* eslint-disable jsx-a11y/media-has-caption */
import './MovieTrailer.scss';
import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getMovieTrailer } from '../../states/trailer/action';
import { getMovieTrailerUrl } from '../../states/trailer/selectors';
import BackButton from '../../sharedComponents/BackButton/BackButton';


const mapDispatch = dispatch => ({
	getMovieTrailer: id => dispatch(getMovieTrailer(id)),
});
const mapState = state => ({
	MovieTrailerUrl: getMovieTrailerUrl(state),
});
const MovieTrailer = (props) => {
	const {
		getMovieTrailer, MovieTrailerUrl, history, match: { params: { id: movieId = '' } },
	} = props;

	useEffect(() => {
		getMovieTrailer(movieId);
	}, movieId);
	const onVideoBack = () => history.goBack();
	return (
		<div className="movie-trailer">
			<BackButton title="Go details" onClick={onVideoBack} />
			<video width="100%" key={MovieTrailerUrl} height="100%" controls autoPlay onEnded={onVideoBack}>
				<source src={MovieTrailerUrl} type="video/mp4" />
			</video>
		</div>
	);
};

MovieTrailer.propTypes = {
	getMovieTrailer: PropTypes.func.isRequired,
	MovieTrailerUrl: PropTypes.string.isRequired,
};

export default connect(mapState, mapDispatch)(MovieTrailer);
