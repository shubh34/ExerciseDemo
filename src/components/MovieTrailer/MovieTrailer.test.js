import React from 'react';
import { shallow } from 'enzyme';

import { MovieTrailer } from './MovieTrailer';
import { trailer } from '../../../tests-stub/Trailer';
import { getMovieTrailerUrl } from '../../states/trailer/selectors';
import BackButton from '../../sharedComponents/BackButton/BackButton';


const setUp = (setUpProps = {}, container = shallow) => {
	const state = { trailer };
	const props = {
		movieTrailerUrl: getMovieTrailerUrl(state),
		getMovieTrailer: jest.fn(),

		match: {
			params: {
				id: 'test',
			},
		},
		history: {
			goBack: jest.fn(),
		},
		...setUpProps,
	};
	const wrapper = container(<MovieTrailer {...props} />);
	return { wrapper, props };
};

describe('MovieTrailer', () => {
	it('should render trailor of movie', () => {
		const { wrapper } = setUp();
		expect(wrapper).toMatchSnapshot();
	});
	it('should naviagte to details page', () => {
		const { wrapper, props } = setUp();
		wrapper.find(BackButton).simulate('click');
		expect(props.history.goBack).toBeCalled();
	});
});
