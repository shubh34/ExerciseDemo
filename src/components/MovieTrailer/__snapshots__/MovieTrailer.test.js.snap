// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTrailer should render trailor of movie 1`] = `
<div
  className="movie-trailer"
>
  <BackButton
    onClick={[Function]}
    title="Go details"
  />
  <video
    autoPlay={true}
    controls={true}
    height="100%"
    key="https://prod-stpeter-pmd.verizon.cdn.rakuten.tv/7/c/3/7c38f0f5b4e1bb200ebb4fa4f6386669-mc-0-137-0-0_SD_TRAILER_PAR_1_1/7c38f0f5b4e1bb200ebb4fa4f6386669-mc-0-137-0-0_SD_TRAILER_PAR_1_1.mp4?UeKaKfcJXUl4EZ34s90Y3c_LOwBF17L76uXbxJmZcPVuu1cmxTXM1YPrR_-tWcASBG7kfDEarl4c5pqAY4a6NrlUPLlVP_mzvcWZ7XfF9a0u7wu7h-dlRysR_VEe_-Q0F9osaJIWYtAREIqGMWbT-Zdq2jOn3ubbvd-xJvcr6NgidJa5w9lYas-7kUdzgEF-O9x0Rv4KadmKPVpRU7qNykIaso8RiQQon4NS5iuPAkh8xGvTGq6xCFKupMEFPhZk2PNr96ThUNRIJg=="
    onEnded={[Function]}
    width="100%"
  >
    <source
      src="https://prod-stpeter-pmd.verizon.cdn.rakuten.tv/7/c/3/7c38f0f5b4e1bb200ebb4fa4f6386669-mc-0-137-0-0_SD_TRAILER_PAR_1_1/7c38f0f5b4e1bb200ebb4fa4f6386669-mc-0-137-0-0_SD_TRAILER_PAR_1_1.mp4?UeKaKfcJXUl4EZ34s90Y3c_LOwBF17L76uXbxJmZcPVuu1cmxTXM1YPrR_-tWcASBG7kfDEarl4c5pqAY4a6NrlUPLlVP_mzvcWZ7XfF9a0u7wu7h-dlRysR_VEe_-Q0F9osaJIWYtAREIqGMWbT-Zdq2jOn3ubbvd-xJvcr6NgidJa5w9lYas-7kUdzgEF-O9x0Rv4KadmKPVpRU7qNykIaso8RiQQon4NS5iuPAkh8xGvTGq6xCFKupMEFPhZk2PNr96ThUNRIJg=="
      type="video/mp4"
    />
  </video>
</div>
`;
